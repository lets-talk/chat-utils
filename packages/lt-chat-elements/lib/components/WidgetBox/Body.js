import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


const StyledBodyContainer = styled.div`
  flex: 1 1 auto; /* same as flex: 1 1 auto; */
  position: relative; /* need this to position inner content */
  overflow-y: auto;
`;
StyledBodyContainer.displayName = 'StyledBodyContainer';


const Body = (props) => {
  const {
    children,
  } = props;
  return (
    <StyledBodyContainer>{children}</StyledBodyContainer>
  );
};

Body.propTypes = {
  children: PropTypes.node,
};

Body.defaultProps = {
};

export default Body;
