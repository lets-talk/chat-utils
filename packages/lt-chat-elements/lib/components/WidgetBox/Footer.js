import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const WidgetBoxFooter = (props) => {
  const {
    children,
    height,
    className,
  } = props;

  const StyledFlexFooter = styled.div`
    flex: 0 0 auto;
    flex-basis: ${() => `${height}px`};
  `;
  StyledFlexFooter.displayName = 'StyledFlexFooter';

  return (
    <StyledFlexFooter className={className}>
      {children}
    </StyledFlexFooter>
  );
};

WidgetBoxFooter.propTypes = {
  children: PropTypes.node.isRequired,
  height: PropTypes.number,
  className: PropTypes.string,
};

WidgetBoxFooter.defaultProps = {
  height: 50,
  className: 'WidgetBoxFooter-Container',
};

WidgetBoxFooter.displayName = 'WidgetBoxFooter';

export default WidgetBoxFooter;
