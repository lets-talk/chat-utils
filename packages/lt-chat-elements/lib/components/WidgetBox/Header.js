import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import { headerMenuButton, themeColor, textColor } from '../../utils/style';

const StyledHeaderContainer = styled.div`
  flex: 0 0 auto;
  flex-basis: ${(props) => `${props.height}px`};

  background-color: ${(props) => themeColor(props.theme, 'accent', 'base')};
  color: ${(props) => textColor(props.theme, 'dark', 'primary')};
  border-radius: 10px 10px 0 0;
  text-align: center;
  padding: 15px 20px;

  button {
    color: ${(props) => textColor(props.theme, 'dark', 'primary')};
    ${headerMenuButton('40px')};
  }

  h4 {
    margin: 0;
    font-size: ${(props) => props.theme.typography.classes.body.fontSize};
    line-height: ${(props) => props.theme.typography.classes.body.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  }

  span {
    color: ${(props) => textColor(props.theme, 'dark', 'secondary')};
    font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
    line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightThin};
  }
`;
StyledHeaderContainer.displayName = 'StyledHeaderContainer';

const Header = (props) => {
  const { children, className, height } = props;
  return (
    <StyledHeaderContainer className={className} height={height}>
      {children}
    </StyledHeaderContainer>
  );
};

Header.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  height: PropTypes.number,
};

Header.defaultProps = {
  className: 'LT-Header-Container',
  height: 50,
};

Header.displayName = 'Header';

export default Header;
