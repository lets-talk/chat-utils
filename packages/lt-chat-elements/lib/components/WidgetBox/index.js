import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import WidgetBoxHeader from './Header';
import WidgetBoxFooter from './Footer';
import WidgetBoxBody from './Body';

const StyledWidgetBoxContainer = styled.div`
  border-radius: 10px 10px 0px 0px;
  box-shadow: 0px 0px 8px 1px rgba(0,0,0,0.44);
  
  margin: 0;
  width: 100%;
  height: 100%;

  display: flex;
  flex-direction: column;
          
  background: rgba(255, 255, 255, .1);
`;

const WidgetBox = (props) => {
  const {
    className,
    children,
  } = props;
  return (
    <StyledWidgetBoxContainer className={className}>{children}</StyledWidgetBoxContainer>
  );
};

WidgetBox.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

WidgetBox.defaultProps = {
  className: 'LT-WidgetBox-Container',
};

WidgetBox.Header = WidgetBoxHeader;
WidgetBox.Footer = WidgetBoxFooter;
WidgetBox.Body = WidgetBoxBody;

export default WidgetBox;
