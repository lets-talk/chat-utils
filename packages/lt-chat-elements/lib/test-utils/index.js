// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
