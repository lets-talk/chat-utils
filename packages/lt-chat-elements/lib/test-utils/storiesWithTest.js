import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
