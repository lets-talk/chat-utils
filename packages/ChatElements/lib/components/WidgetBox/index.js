import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import Header from './components/Header';
import MessageList from './components/MessageList';
import Sender from './components/Sender';
import './index.scss';

const WidgetBox = (props) => {
  const {
    messages,
    title,
    subtitle,
    avatar,
    senderPlaceHolder,
    className,
    toggleChat,
    disabledInput,
    openMenu,
    showMinimizeButton,
    showMenuButton,
    sendMessage,
  } = props;

  return (
    <div className={classNames('widgetbox-container', className)}>
      <Header
        title={title}
        subtitle={subtitle}
        avatar={avatar}
        toggleChat={toggleChat}
        openMenu={openMenu}
        showMinimizeButton={showMinimizeButton}
        showMenuButton={showMenuButton}
      />
      <MessageList
        messages={messages}
      />
      <Sender
        sendMessage={sendMessage}
        placeholder={senderPlaceHolder}
        disabledInput={disabledInput}
      />
    </div>
  );
};

WidgetBox.propTypes = {
  messages: PropTypes.array,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  avatar: PropTypes.string,
  className: PropTypes.string,
  sendMessage: PropTypes.func,
  senderPlaceHolder: PropTypes.string,
  toggleChat: PropTypes.func,
  openMenu: PropTypes.func,
  showMinimizeButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
  disabledInput: PropTypes.bool,
};

export default WidgetBox;
