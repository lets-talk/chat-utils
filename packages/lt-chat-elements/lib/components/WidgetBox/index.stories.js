import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text, boolean, object, number, select } from '@storybook/addon-knobs';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
// Component to show on storybook
import { WidgetBox, MessageList } from '../../../lib';

const stories = storiesOf('WidgetBox', module);
stories.addDecorator(withKnobs);

const agent = {
  avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
  email: '',
  type: 'Agent',
  status: 'online',
};

stories.addWithInfo(
  'Basic Usage - Empty WidgetBox',
  'This is the basic usage of a Widget Box.',
  () => (
    <div
      style={{
        position: 'fixed', bottom: 0, right: '10px', width: 'calc(100% - 20px)', height: '100%',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <WidgetBox>
          <WidgetBox.Header
            title={text('Title', 'Ayuda en linea')}
            subtitle={text('Subtitle', '')}
            toggleChat={action('toggleChat')}
            openMenu={action('openMenu')}
            showMenuButton={boolean('Show Menu Button', true)}
            showMinimizeButton={boolean('Show Minimize Button', true)}
            height={number('Height', 50)}
            person={object('Person', agent)}
          />
          <WidgetBox.Body>
            <MessageList
              messages={[]}
              onActionMessageClick={action('onActionMessageClick')}
            />
          </WidgetBox.Body>
        </WidgetBox>
      </WrapWithTheme>
    </div>
  )
);
