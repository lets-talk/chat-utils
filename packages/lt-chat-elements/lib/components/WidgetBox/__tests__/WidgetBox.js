import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import WidgetBox from '../index';
import { mountWithTheme } from '../../../test-utils';

describe('WidgetBox component', () => {
  it('should render with no props', () => {
    const component = shallow(<WidgetBox />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    /* eslint-disable react/jsx-closing-tag-location */
    const component = shallow(<WidgetBox>
      <WidgetBox.Header>
        <div>Header Child</div>
      </WidgetBox.Header>
      <WidgetBox.Body><span>Child WidgetBox Body</span></WidgetBox.Body>
    </WidgetBox>);
    /* eslint-disable react/jsx-closing-tag-location */

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();

    expect(component.find('Header').length).toBe(1);
    expect(component.find('Body').length).toBe(1);
    expect(component.find('Body').children().text()).toBe('Child WidgetBox Body');
  });

  it('should mount the component. The component must include Header, Body and Footer', () => {
    const component = mountWithTheme(<WidgetBox>
      <WidgetBox.Header>
        <div>Header Child</div>
      </WidgetBox.Header>
      <WidgetBox.Body>
        <span>Child WidgetBox Body</span>
      </WidgetBox.Body>
      <WidgetBox.Footer>
        <span>I am a footer</span>
      </WidgetBox.Footer>
    </WidgetBox>);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();

    expect(component.find('Header').length).toBe(1);
    expect(component.find('Footer').length).toBe(1);
    expect(component.find('Body').length).toBe(1);
    expect(component.find('Body').children().text()).toBe('Child WidgetBox Body');
  });
});
