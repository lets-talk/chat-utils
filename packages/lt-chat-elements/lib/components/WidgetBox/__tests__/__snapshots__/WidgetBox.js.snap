// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetBox component should mount the component. The component must include Header, Body and Footer 1`] = `
<WidgetBox
  className="LT-WidgetBox-Container"
>
  <styled.div
    className="LT-WidgetBox-Container"
  >
    <div
      className="LT-WidgetBox-Container sc-htpNat kwWgIV"
    >
      <WidgetBoxHeader
        className="WidgetBoxHeader-Container"
        height={50}
      >
        <StyledHeaderContainer
          className="WidgetBoxHeader-Container"
          height={50}
        >
          <div
            className="WidgetBoxHeader-Container sc-bdVaJa dFfQGU"
            height={50}
          >
            <div>
              Header Child
            </div>
          </div>
        </StyledHeaderContainer>
      </WidgetBoxHeader>
      <WidgetBoxBody
        className="WidgetBoxBody-Container"
      >
        <StyledBodyContainer
          className="WidgetBoxBody-Container"
        >
          <div
            className="WidgetBoxBody-Container sc-bwzfXH bQmfDh"
          >
            <span>
              Child WidgetBox Body
            </span>
          </div>
        </StyledBodyContainer>
      </WidgetBoxBody>
      <WidgetBoxFooter
        className="WidgetBoxFooter-Container"
        height={50}
      >
        <StyledFlexFooter
          className="WidgetBoxFooter-Container"
        >
          <div
            className="WidgetBoxFooter-Container sc-bxivhb hVXoeT"
          >
            <span>
              I am a footer
            </span>
          </div>
        </StyledFlexFooter>
      </WidgetBoxFooter>
    </div>
  </styled.div>
</WidgetBox>
`;

exports[`WidgetBox component should render with custom props 1`] = `
<styled.div
  className="LT-WidgetBox-Container"
>
  <WidgetBoxHeader
    className="WidgetBoxHeader-Container"
    height={50}
  >
    <div>
      Header Child
    </div>
  </WidgetBoxHeader>
  <WidgetBoxBody
    className="WidgetBoxBody-Container"
  >
    <span>
      Child WidgetBox Body
    </span>
  </WidgetBoxBody>
</styled.div>
`;

exports[`WidgetBox component should render with no props 1`] = `
<styled.div
  className="LT-WidgetBox-Container"
/>
`;
